kedro-datasets/kedro_datasets/api/api_dataset.py

```diff
@@ -222,13 +234,52 @@ def _execute_save_request(self, json_data: Any) -> requests.Response:
     def save(self, data: Any) -> requests.Response:  # type: ignore[override]
         if self._request_args["method"] in ["PUT", "POST"]:
             if isinstance(data, list):
-                return self._execute_save_with_chunks(json_data=data)
-
-            return self._execute_save_request(json_data=data)
+                response: requests.Response = self._execute_save_with_chunks(json_data=data)
+            else:
+                response: requests.Response = self._execute_save_request(json_data=data)
+
+            if self._wrapped_dataset is None:
+                return response
+            if self._extension == "json":
+                self.wrapped_dataset.save(response.json()) #TODO(npfp): expose json loads arguments
+            elif self._extension == "text":
+                self.wrapped_dataset.save(response.text)
+            elif self._extension:
+                self.wrapped_dataset.save(response)
```

Comment:
Might be better to check on the `self.wrapped_dataset` type:
```suggestion
            elif isinstance(self.wrapped_dataset, JSONDataset):
                self.wrapped_dataset.save(response.json()) #TODO(npfp): expose json loads arguments
            elif  isinstance(self.wrapped_dataset, TextDataset)
                self.wrapped_dataset.save(response.text)
            elif self._extension:
                self.wrapped_dataset.save(response)
```